From 14f2347c5d8978865eb3748724e5c2e3d7fae4ea Mon Sep 17 00:00:00 2001
From: Gaetan Lepage <gaetan@glepage.com>
Date: Mon, 22 Jan 2024 23:53:39 +0100
Subject: [PATCH] pyproject.toml: remove unknown properties

---
 pyproject.toml | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 05461e4a..5b1d9971 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,12 +3,3 @@ first_party_detection = false
 
 [build-system]
 requires = ["setuptools", "wheel", "torch", "ninja"]
-
-first_party_detection = false
-
-target-version = ["py38"]
-
-excludes = [
-    "gallery",
-    "tutorials",
-]
-- 
2.43.0

