diff --git a/data/tests/meson.build b/data/tests/meson.build
index 3da184010..29df69eb7 100644
--- a/data/tests/meson.build
+++ b/data/tests/meson.build
@@ -105,7 +105,7 @@ configure_file(
   output: 'fwupd-tests.conf',
   configuration: con2,
   install: true,
-  install_dir: join_paths(datadir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(get_option('installed_test_prefix'), 'etc', 'fwupd', 'remotes.d'),
 )
 
 if umockdev_integration_tests.allowed()
diff --git a/meson.build b/meson.build
index d590b3a72..292a7be3d 100644
--- a/meson.build
+++ b/meson.build
@@ -193,8 +193,8 @@ else
   datadir = join_paths(prefix, get_option('datadir'))
   sysconfdir = join_paths(prefix, get_option('sysconfdir'))
   localstatedir = join_paths(prefix, get_option('localstatedir'))
-  installed_test_bindir = join_paths(libexecdir, 'installed-tests', meson.project_name())
-  installed_test_datadir = join_paths(datadir, 'installed-tests', meson.project_name())
+  installed_test_bindir = join_paths(get_option('installed_test_prefix'), 'libexec', 'installed-tests', meson.project_name())
+  installed_test_datadir = join_paths(get_option('installed_test_prefix'), 'share', 'installed-tests', meson.project_name())
   daemon_dir = join_paths(libexecdir, 'fwupd')
 endif
 mandir = join_paths(prefix, get_option('mandir'))
@@ -546,6 +546,7 @@ gnome = import('gnome')
 i18n = import('i18n')
 
 conf.set_quoted('FWUPD_PREFIX', prefix)
+conf.set_quoted('FWUPD_INSTALLED_TEST_PREFIX', get_option('installed_test_prefix'))
 conf.set_quoted('FWUPD_BINDIR', bindir)
 conf.set_quoted('FWUPD_LIBDIR', libdir)
 conf.set_quoted('FWUPD_LIBEXECDIR', libexecdir)
diff --git a/meson_options.txt b/meson_options.txt
index 3c021d52a..195ee8ba5 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -329,6 +329,10 @@ option('systemd_unit_user',
   value: 'fwupd-refresh',
   description: 'User account to use for fwupd-refresh.service (empty for DynamicUser)',
 )
+option('installed_test_prefix',
+  type: 'string',
+  description: 'Prefix for installed tests'
+)
 option('tests',
   type: 'boolean',
   value: true,
